### PR TITLE
docs: guidelines

### PR DESCRIPTION
**Pull request by enitrat**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #<Issue number>

## What is the new behavior?

-
-
-

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1188)

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -5,67 +5,133 @@ to make via issue, before making a change. Please note we have a
...
+- If you have an idea for a new feature or a bug fix, please create an issue
+  first. This will allow us to discuss the idea and provide feedback before you
+  start working on it.
+- If you are working on an issue, please ask to be assigned. This will help us
+  keep track of who is working on what and avoid duplicated work.
+- Do not ask to be assigned to an issue if you are not planning to work on it
+  immediately. We want to keep the project backlog clean and organized. If you
+  are interested in working on an issue, please comment on the issue and we will
+  assign it to you. We will remove the assignment if you do not start working on
+  the issue within a 2 days. You can, of course, submit a draft PR if you need
```
Comment (ClementWalter) [inline docs/CONTRIBUTING.md:22]: typo "within a 2 days"

Comment (enitrat) [reply docs/CONTRIBUTING.md:22]:
```suggestion
  the issue within 2 days. You can, of course, submit a draft PR if you need
```

Comment (ClementWalter) [inline docs/CONTRIBUTING.md:22]:
```suggestion
  the issue within 2 days. You can, of course, submit a draft PR if you need
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -5,67 +5,133 @@ to make via issue, before making a change. Please note we have a
...
+  starknet-compile binary) to `starknet-compile` in order to make the CairoLS
+  VSCode extension work.
+
+  ```bash
+  ln -s <YOUR_PATH_TO_YOUR_PYTHON_VENV_BINARIES>/starknet-compile-deprecated <YOUR_PATH_TO_LOCAL_BINARIES>/starknet-compile
+  # example: ln -s /Users/eliastazartes/code/kakarot/.venv/bin/starknet-compile-deprecated /usr/local/bin/starknet-compile
+  ```
+
+  If you followed the instructions above, you can run
+  `poetry show cairo-lang -v` to find the path to your cairo-lang dependency,
```
Comment (ClementWalter) [inline docs/CONTRIBUTING.md:123]: starknet-compile and not cairo-lang ?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -5,67 +5,133 @@ to make via issue, before making a change. Please note we have a
...
+  and the binaries should be in the `bin` directory.
+
+- python3.10 is not compatible with the cairo-lang library. Make sure poetry and
+  your pyenv are using the 3.10 version of Python. Your machine may have
+  conflicting versions of python.
```
Comment (ClementWalter) [inline docs/CONTRIBUTING.md:128]: not sure about this but the last part "your machine may have conflicting versions of python." often when people mix conda, brew and other installers. The solution is "just remove everything and use pyenv" see also https://stackoverflow.com/questions/51125013/how-can-i-install-a-previous-version-of-python-3-in-macos-using-homebrew/60189988#60189988

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (enitrat) [reply docs/CONTRIBUTING.md:123]: no, `cairo-lang` shows where the package is installed, and the `starknet-compile` binary will be under `bin/starknet-compile-deprecated`

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -5,67 +5,133 @@ to make via issue, before making a change. Please note we have a
...
+- Install [docker](https://docs.docker.com/get-docker/) to build some
+  experimental solidity contracts that require a custom solc compiler
+- Install [foundry](https://book.getfoundry.sh/getting-started/installation) to
+  compile the Solidity contracts we use for testing.
+- Install
+  [scarb using asdf](https://docs.swmansion.com/scarb/download.html#install-via-asdf)
+  to the Cairo contracts we use as dependencies. As these dependencies rely on
+  different versions of Cairo, you will need to install the following versions
+  of Scarb:
+
```
Comment (ClementWalter) [inline docs/CONTRIBUTING.md:52]: and go for pprof for printing profile files

**Pull request merged**